src/Miner/GetConfig.cpp

```diff
@@ -437,19 +441,45 @@ Pool getPool()
 
         std::getline(std::cin, algorithm);
 
+        int algorithmNumber;
+        bool algorithmIsNumber;
+
         if (algorithm == "")
         {
             continue;
         }
 
+        try
+        {
+            algorithmNumber = std::stoi(algorithm);
+            algorithmIsNumber = true;
+        }
+        catch (const std::invalid_argument &)
+        {
+            algorithmIsNumber = false;
+        }
+
+        if (algorithmIsNumber) 
+        {
+            auto selectedAlgorithm = availableAlgorithms.find(algorithmNumber); // finding if item is present in map
+            if (selectedAlgorithm != availableAlgorithms.end())
+            {
+                algorithm = selectedAlgorithm->second;
+            }
+            else
+            {
+                std::cout << WarningMsg("Unknown algorithm \"" + std::to_string(algorithmNumber) + "\". Try again.") << std::endl;
```

Comment:
Missing a `continue` under this line.

Also, this message might be slightly more informative.

```suggestion
                std::cout << WarningMsg("Bad input, expected an algorithm name, or number from ") << InformationMsg("1") << WarningMsg(" to ") << InformationMsg(availableAlgorithms.size()) << std::endl;
```